src/controllers/save.and.resume.controller.ts

```diff
@@ -0,0 +1,83 @@
+
+import * as config from "../config";
+
+import { ApplicationData } from "../model";
+import { createAndLogErrorRequest, logger } from "../utils/logger";
+import { setExtraData } from "../utils/application.data";
+import { isActiveFeature } from "../utils/feature.flag";
+import { resumeOverseasEntity } from "../service/overseas.entities.service";
+import { HasSoldLandKey, ID, IsSecureRegisterKey, OverseasEntityKey, Transactionkey } from "../model/data.types.model";
+import { WhoIsRegisteringKey, WhoIsRegisteringType } from "../model/who.is.making.filing.model";
+import { OverseasEntityDueDiligence, OverseasEntityDueDiligenceKey } from "../model/overseas.entity.due.diligence.model";
+import { DueDiligence, DueDiligenceKey } from "../model/due.diligence.model";
+import { BeneficialOwnerGov, BeneficialOwnerGovKey } from "../model/beneficial.owner.gov.model";
+import { BeneficialOwnerIndividual, BeneficialOwnerIndividualKey } from "../model/beneficial.owner.individual.model";
+import { BeneficialOwnerOther, BeneficialOwnerOtherKey } from "../model/beneficial.owner.other.model";
+import { ManagingOfficerCorporate, ManagingOfficerCorporateKey } from "../model/managing.officer.corporate.model";
+import { ManagingOfficerIndividual, ManagingOfficerKey } from "../model/managing.officer.model";
+import { Session } from "@companieshouse/node-session-handler";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `GET Save And Resume`);
+
+    const { transactionId, overseaEntityId } = req.params;
+    const infoMsg = `Transaction ID: ${transactionId}, OverseasEntity ID: ${overseaEntityId}`;
+
+    logger.infoRequest(req, `Resuming OE - ${infoMsg}`);
+
+    if (isActiveFeature(config.FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022)) {
+      const appData: ApplicationData = await resumeOverseasEntity(req, transactionId, overseaEntityId);
+
+      if (!appData || !Object.keys(appData).length) {
+        throw createAndLogErrorRequest(req, `Error on resuming OE - ${infoMsg}`);
+      }
+
+      setWebApplicationData(req.session as Session, appData, transactionId, overseaEntityId);
+    }
+
+    return res.redirect(config.SOLD_LAND_FILTER_URL);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+/**
+ * Set default values needed for the web journey that are not part of OE API data model
+ * nor part of the SDK mapper object.
+ * Add IDs needed for the web to select single BOs or MOs on beneficial-owner-type screen and
+ * related checkboxes for different pages
+ * @param session
+ * @param appData
+ * @param transactionId
+ * @param overseaEntityId
+ */
+const setWebApplicationData = (session: Session, appData: ApplicationData, transactionId: string, overseaEntityId: string) => {
+
+  appData[BeneficialOwnerIndividualKey] = (appData[BeneficialOwnerIndividualKey] as BeneficialOwnerIndividual[])
+    .map( e => { return { ...e, [ID]: uuidv4() }; } );
```

Comment:
won't the uuid4() generate a different uuid to the one returned from mongo?

Reply:
Hi, mongo does not return IDs for any OE object even BOs or MOs

Reply:
ah ok, i forgot we use them for adding/removing in the web app only.